### ISSUE TITLE
[FR] Optmizer / Auto Option

### ISSUE THREAD

**Issue by davidhellmann** [engram-design/ImageResizer#56]
Two FRs :)

1. Would be nice to use some optimizers when the image transforms to get smaller file sizes

2. We had a problem when the Client uploads long images with 4000x10000px — it would be nice to set an auto option like: 
```
width: 3000px
heigh: auto
```

Cheers!

**Comment by engram-design** [comment]
For the second point, do you mean you _want_ a 10000px high image, so you want it set to `auto`? As opposed to say 4000px x 4000px restrictions?

**Comment by davidhellmann** [comment]
No, sometime the client have Images or more screenshots where the size is like 2000x8000px
Now, when I set the image resizer to 1000x1000px I get an image with: 250x1000px and that's not that what I need. It would be nice to only set the X-Axis max width. 

1000px x AUTO to become an image with: 1000x4000px and that would be fine. But the 250px image is completely useless.

Hope now it's more clear.

**Comment by engram-design** [comment]
Gotcha, makes sense! I'll look at adding that support in.
